library/network/src/lib/network/firewalld.rb

```diff
@@ -140,31 +141,31 @@ def zones
     # @param zone [String] The firewall zone
     # @return [Array<String>] list of zone's interfaces
     def list_interfaces(zone)
-      fwd_result("--permanent --zone=#{zone} --list-interfaces").split(" ")
+      fwd_result("--permanent --zone=#{zone.shellescape} --list-interfaces").split(" ")
```

Comment:
I know that this is beyond the scope of this PR, but still...

It appears to me that the result of this `fwd_result` function is _always_ sent straight away to `split()` to be split up into pieces. So why not change that function so it returns the array that everybody using it needs?

Reply:
No. In this PR alone I can see `.split " "`, `.split "\n"`, `.strip` and `.rstrip` applied on it. QED.